### DISCUSSION TITLE
Use one pool for two or more orgs

### DISCUSSION THREAD

**Discussion by adeepn** [cloudbase/garm#176] [Ideas]
How about using one pool for multiple organizations?

**Comment by gabriel-samfira** [comment]
Hi @adeepn !

There is no way to support that. A pool will register a runner within an org and make it available to repositories under that org. There is no way for a single pool to register runners in two different orgs. You will either have to create one pool per org, or define the pool at the enterprise level (if both orgs are under the same enterprise).

**Reply by adeepn** [reply]
hmm. we can interpret min-runner count as min for every connected org for this moment.
Ok, what about limit max runner count per provider?

**Reply by gabriel-samfira** [reply]
The model is as follows:

Entity (repo, org, enterprise) has multiple pools, and has an associated webhook installed in Github.
Pools have multiple runners of the same type
Pools also have a 1:1 relation to a provider.
A provider may have multiple pools leveraging it.
A provider is statically defined in the config.


There is no way to limit runners per provider. You can only set a had cap per pool. Considering that multiple pools may leverage a provider, there would need to be some mechanism by which pools could contend for the resources of the provider. That is a level of granularity that does not make sense in garm. Especially because that may introduce situations where a pool may wait a long time for the provider to have resources available to spin up a runner, making that pool largely useless. It leads to poor user experience.

What you can do is configure a single pool per provider. That way you can decide how many runners get created in that provider. You can define multiple providers with credentials for the same cloud if you wish, and set each individual provider on a single pool.

Ideally, the provider you want to configure for your pools, has enough resources to handle multiple pools.
